### ISSUE TITLE
Doc for 'sources' seems not accurate

### ISSUE THREAD

**Issue by jingsam** [mapbox/mapbox-gl-style-spec#526]
https://github.com/mapbox/mapbox-gl-style-spec/blob/mb-pages/reference/v8.json#L142-L147

```
  "sources": {
    "*": {
      "type": "source",
      "doc": "Specification of a data source. For vector and raster sources, either TileJSON or a URL to a TileJSON must be provided. For GeoJSON and video sources, a URL must be provided."
    }
  },
```

For geojson sources, we can also use inline data property. For image sources, a URL must be provided.

**Comment by jfirebaugh** [comment]
Correct.  I don't think this doc string is actually used anywhere however.

**Comment by jfirebaugh** [comment]
Fixed in #531.
